### PR TITLE
Update to latest master of OpenSCAD.

### LINKED ISSUE #37: Add PulseAudio permissions by default
Otherwise the sound notification feature (which is enabled by default) might end up generating some warnings like:
```
May 29 22:34:14 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:18 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:29 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:32 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:36 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:38 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:38 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:39 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:39 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:44 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
May 29 22:34:48 classic org.openscad.OpenSCAD.desktop[105145]: PulseAudioService: pa_context_connect() failed
```
etc.

I left OpenSCAD running overnight and:
```sh
$ journalctl -b | grep "pa_context_connect() failed" | wc -l
185765
```

See also https://github.com/openscad/openscad/issues/4921

### PR DESCRIPTION
**Pull request by t-paul**
Fixes #37.

### CODE REVIEW

**Comment by hadess** [comment]
The error is:
```
Run failed, some data was ignored.
Errors were raised during this compose run:
org.openscad.OpenSCAD
  E: no-stock-icon
```

I think that [this](https://github.com/openscad/openscad/blob/master/openscad.appdata.xml.in#L7) might be the reason:
```xml
  <icon type="remote">https://www.openscad.org/images/openscad${SNAPSHOT_SUFFIX}.png</icon>
```

Removing that line should be enough.

**Comment by t-paul** [comment]
Going to try `<icon type="stock">openscad</icon>` first. If that does not work, I can try removing all icon references but that would be a flathub hack which would be annoying.

**Comment by hfiguiere** [comment]
icon stock is worse.

**Comment by t-paul** [comment]
How, why? Removing the remote icon will very likely break other places, e.g. the snap store.

**Comment by hadess** [comment]
I meant that you could remove the line in a Flathub patch, or a `sed` call.

**Comment by hfiguiere** [comment]
appstream compose will make the catalog data using the .desktop and the icon it references. frillions of packages do that (no icon in the appstream file) and not one say "it doesn't work on snap") (I have no idea about snap)

**Comment by t-paul** [comment]
What I don't get is the reason for this.

1) The error message says "there is no stock item"
2) Looking around on github, I find [many projects](https://github.com/search?q=%3Cicon+type%3D%22stock%22%3E&type=code) having stock items in the metainfo
3) [Spec](https://www.freedesktop.org/software/appstream/docs/chap-CatalogData.html#tag-ct-icon) explains stock icons as valid option

Now I get that the definition of a stock item is a bit muddy in context of flatpak, is that stock item folder inside or outside of the packaging. But then why does it error out if it's not set? And why would it force extra hacks if the normal application repo has it specified? And why is removing *all* icons the solution?

**Comment by hadess** [comment]
> Looking around on github, I find [many projects](https://github.com/search?q=%3Cicon+type%3D%22stock%22%3E&type=code) having stock items in the metainfo

Looks like none of them are `type=remote`, and networking isn't available at this stage of the build. A better error message would be nice, but that's probably the reason why there's one in the first place.

File: org.openscad.OpenSCAD.yaml [hunk 1]
```diff
@@ -134,26 +150,56 @@ modules:
...
+      - -DUSE_BUILTIN_OPENCSG=ON
     cleanup:
       - /share/pixmaps
     sources:
       - type: git
         url: https://github.com/openscad/openscad.git
-        commit: 05b962168cce9bc30b967a19514c490b495bd732
+        commit: 6db5adc0c4fbe825eab7e839ff9617e7806e4e38
     post-install:
       - mv /app/share/applications/openscad.desktop /app/share/applications/openscad-nightly.desktop
```
Comment (hfiguiere) [inline org.openscad.OpenSCAD.yaml:204]: why moving to do a "rename-desktop-file" after?

File: org.openscad.OpenSCAD.yaml [hunk 2]
```diff
@@ -134,26 +150,56 @@ modules:
...
       - mv /app/share/mime/packages/openscad.xml /app/share/mime/packages/org.openscad.OpenSCAD.xml
```
Comment (hfiguiere) [inline org.openscad.OpenSCAD.yaml:205]: `rename-mime-file` now exist for this.

**Comment by hfiguiere** [comment]
But are you sure that the icon defined in the desktop file is found? `Icon=openscad${SNAPSHOT_SUFFIX}`

File: org.openscad.OpenSCAD.yaml (hunk 2 above)
Comment (hfiguiere) [reply org.openscad.OpenSCAD.yaml:205]: it's done in the `master` branch already

**Comment by hfiguiere** [comment]
but if you really want to solve the problem, rename everything upstream to follow the app ID convention (icon, appstream file, desktop file, mime file). It works everywhere and make things simpler.

File: org.openscad.OpenSCAD.yaml [hunk 3]
```diff
@@ -3,8 +3,10 @@ runtime: org.kde.Platform
 runtime-version: 5.15-23.08
 sdk: org.kde.Sdk
 command: openscad
-rename-desktop-file: openscad-nightly.desktop
 rename-icon: openscad-nightly
+rename-mime-file: openscad.xml
+rename-desktop-file: openscad.desktop
+rename-appdata-file: openscad.appdata.xml
```
Comment (hfiguiere) [inline org.openscad.OpenSCAD.yaml:9]: according to this https://github.com/openscad/openscad/blob/master/CMakeLists.txt#L1315 the appstream file is already renamed to something else.

File: org.openscad.OpenSCAD.yaml [hunk 4]
```diff
@@ -3,8 +3,10 @@ runtime: org.kde.Platform
 runtime-version: 5.15-23.08
 sdk: org.kde.Sdk
 command: openscad
-rename-desktop-file: openscad-nightly.desktop
 rename-icon: openscad-nightly
+rename-mime-file: openscad.xml
```
Comment (hfiguiere) [inline org.openscad.OpenSCAD.yaml:7]: there is also a rename-mime-icon like used on the master branch. Should replace line 206 (deleted)

File: org.openscad.OpenSCAD.yaml (hunk 3 above)
Comment (t-paul) [reply org.openscad.OpenSCAD.yaml:9]: Right, I thought I'm looking at those renames in the logs, but I guess instead it was that `make install` instead... tbc... at some later time. Thanks!

File: org.openscad.OpenSCAD.yaml (hunk 4 above)
Comment (t-paul) [reply org.openscad.OpenSCAD.yaml]: Argh, :man_facepalming:. I did not have the master branch updated to the latest changes.

Locally I'm not getting a "missing screenshots" error which is a bit confusing again, lets see what the official builder has to say to that (all screenshots are remote URLs right now).

**Pull request merged**